shell/browser/native_window_mac.mm

```diff
@@ -1810,6 +1810,22 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   [window_ setAcceptsMouseMovedEvents:forward];
 }
 
+gfx::Rect NativeWindowMac::GetWindowControlsOverlayRect() {
+  gfx::Rect bounding_rect;
+  if (title_bar_style_ == TitleBarStyle::kHidden ||
+      title_bar_style_ == TitleBarStyle::kHiddenInset) {
```

Comment:
The `titleBarStyle` is simply a preset of `frame: false` and `setWindowButtonVisibility`/`setTrafficLightPosition` calls, and it is possible for a normal frameless window to have custom window buttons by calling `win.setWindowButtonVisibility`.

So the condition should be something like `if (!frame() && buttons_view_ && ![buttons_view_ hidden])`.

Reply:
@zcbenz ok I will look into this.  There is a problem with setTrafficLightPosition because Windows Control Overlay calculates and communicates the rectangle of available space in the title bar but this is based on the assumption that the traffic lights are either on the left or right of the screen.  If someone has a strange position for these controls then the calculations/rectangle will not be accurate

Reply:
@zcbenz I made the suggested changes.  When trafficLightPosition is set, the Windows Control Overlay will return an empty rectangle because it is indeterminate what the Windows Control Overlay region should be.  Also, if app developers are moving traffic lights to specific positions, then they already have the information needed to determine the screen area available for custom title bars.